public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -384,6 +385,22 @@ export class LokiDatasource
     return res.data || [];
   }
 
+  async queryStatsRequest(query: LokiQuery): Promise<QueryStats> {
+    const labelMatcher = `{${query.expr.split('{')[1].split('}')[0]}}`;
+    const { start, end } = this.getTimeRangeParams();
+
+    const { data } = await lastValueFrom(
+      getBackendSrv().fetch({
+        method: 'GET',
+        headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+        params: { query: labelMatcher, start, end },
+        url: `${this.instanceSettings.url}/loki/api/v1/index/stats`,
+      })
+    );
+
+    return data as QueryStats;
```

Comment:
Instead of this `as` assignment, you can also add a generic to the `fetch` call: 

```
getBackendSrv().fetch<QueryStats>()
```